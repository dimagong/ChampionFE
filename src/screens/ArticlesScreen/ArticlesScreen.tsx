import * as React from 'react'

import {
    View,
    StatusBar,
    SafeAreaView,
    StyleSheet,
    ScrollView,
} from 'react-native'
import {CardComponent} from '@ui/components'

export const ArticlesScreen = ({
    route,
    navigation,
}: {
    route: any
    navigation: any
}) => {
    const {title, subTitle, content, url, id} = route.params

    return (
        <SafeAreaView style={styles.container}>
            <StatusBar barStyle="light-content" backgroundColor="#000000" />
            <ScrollView>
                <View style={styles.container__view}>
                    <CardComponent
                        id={id}
                        url={url}
                        title={title}
                        content={content}
                        subTitle={subTitle}
                    />
                </View>
            </ScrollView>
        </SafeAreaView>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    container__view: {
        padding: 20,
    },
})
