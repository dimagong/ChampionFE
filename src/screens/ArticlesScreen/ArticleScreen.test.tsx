import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render, screen, fireEvent} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import {ArticlesScreen} from './ArticlesScreen';

const mockFn = jest.fn();
const component = (
  <ArticlesScreen
    route={{
      params: {
        title: 'The best players of the previous game',
        content: 'Junior players received the best marks',
        img: 'ui/images/reward.jpg',
      },
    }}
    navigation={mockFn}
  />
);

describe('Testing ArticleScreen', () => {
  test('Renders correctly', () => {
    const content = renderer.create(component).toJSON();
    expect(content).toMatchSnapshot();
  });
  test('Page contains properly messages', () => {
    render(component);
    const contentPlayer = screen.getByText(
      'Junior players received the best marks',
    );
    const contentTitle = screen.getByText(
      'The best players of the previous game',
    );
    expect(contentPlayer).toBeTruthy();
    expect(contentTitle).toBeTruthy();
  });
});
