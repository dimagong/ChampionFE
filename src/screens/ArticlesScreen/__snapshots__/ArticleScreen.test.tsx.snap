// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ArticleScreen Renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "padding": 20,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "alignSelf": undefined,
              "bottom": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(255, 251, 254, 1)",
                  "borderRadius": 4,
                  "elevation": 1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "disabled": true,
                  }
                }
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "flexGrow": 1,
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "#eeeeee",
                        "height": 195,
                        "overflow": "hidden",
                      },
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <Image
                    accessibilityIgnoresInvertColors={true}
                    siblings={
                      [
                        "withTheme(Card.Cover)",
                        "Card.Content",
                      ]
                    }
                    style={
                      [
                        {
                          "flex": 1,
                          "height": undefined,
                          "justifyContent": "flex-end",
                          "padding": 16,
                          "width": undefined,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  style={
                    [
                      {
                        "paddingHorizontal": 16,
                      },
                      {
                        "paddingBottom": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        {
                          "textAlign": "left",
                        },
                        [
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 0.87)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "fontSize": 20,
                              "letterSpacing": 0.15,
                              "lineHeight": 30,
                              "marginVertical": 2,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    The best players of the previous game
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        {
                          "textAlign": "left",
                        },
                        [
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 0.87)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "fontSize": 14,
                              "letterSpacing": 0.25,
                              "lineHeight": 20,
                              "marginVertical": 2,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    Junior players received the best marks
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Testing ArticleScreen Rrenders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "padding": 20,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "alignSelf": undefined,
              "bottom": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(255, 251, 254, 1)",
                  "borderRadius": 4,
                  "elevation": 1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "disabled": true,
                  }
                }
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "flexGrow": 1,
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "#eeeeee",
                        "height": 195,
                        "overflow": "hidden",
                      },
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <Image
                    accessibilityIgnoresInvertColors={true}
                    siblings={
                      [
                        "withTheme(Card.Cover)",
                        "Card.Content",
                      ]
                    }
                    style={
                      [
                        {
                          "flex": 1,
                          "height": undefined,
                          "justifyContent": "flex-end",
                          "padding": 16,
                          "width": undefined,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  style={
                    [
                      {
                        "paddingHorizontal": 16,
                      },
                      {
                        "paddingBottom": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        {
                          "textAlign": "left",
                        },
                        [
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 0.87)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "fontSize": 20,
                              "letterSpacing": 0.15,
                              "lineHeight": 30,
                              "marginVertical": 2,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    The best players of the previous game
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        {
                          "textAlign": "left",
                        },
                        [
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 0.87)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "fontSize": 14,
                              "letterSpacing": 0.25,
                              "lineHeight": 20,
                              "marginVertical": 2,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    Junior players received the best marks
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
